yarn-project/key-store/src/test_key_store.ts

```diff
@@ -102,6 +109,20 @@ export class TestKeyStore implements KeyStore {
     return Promise.resolve(Point.fromBuffer(masterNullifierPublicKeyBuffer));
   }
 
+  /**
+   * Gets the master nullifier public key for a given master nullifier public key hash.
+   * @throws If the master nullifier public key hash does not exist in the key store.
+   * @param masterNullifierPublicKeyHash - The master nullifier public key hash for which to retrieve the master nullifier public key.
+   * @returns The master nullifier public key for the account.
+   */
+  getMasterNullifierPublicKeyWithMasterNullifierPublicKeyHash(masterNullifierPublicKeyHash: Fr): Promise<PublicKey> {
```

Comment:
I feel like this is not the way to do it. Please just create a utility function which fetches all the accounts, based on that fetches all the nullifier public keys, hashes them and gets the address based on that. There will never bee a ton of accounts in the store so the inefficiency is fine. If this turns out to be too common we can consider changing the key store.